freeipa/src/main/java/com/sequenceiq/freeipa/service/resource/ResourceAttributeUtil.java

```diff
@@ -1,25 +1,41 @@
 package com.sequenceiq.freeipa.service.resource;
 
 import java.io.IOException;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
-import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.VolumeSetAttributes;
 import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.common.json.Json;
 import com.sequenceiq.freeipa.entity.Resource;
 
 @Component
 public class ResourceAttributeUtil {
 
-    public <T> Optional<T> getTypedAttributes(Resource resource, Class<T> attributeType) {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceAttributeUtil.class);
+
+    public Optional<Object> getTypedAttributes(Resource resource) {
         Json attributes = resource.getAttributes();
+        Optional<Object> ret = Optional.empty();
         try {
-            return Objects.nonNull(attributes.getValue()) ? Optional.ofNullable(attributes.get(attributeType)) : Optional.empty();
+            if (Objects.nonNull(attributes.getValue())) {
+                Map<String, Object> map = attributes.get(Map.class);
+                String className = map.getOrDefault(CloudResource.ATTRIBUTE_TYPE, VolumeSetAttributes.class.getCanonicalName()).toString();
```

Comment:
We have another Attribute class(`EIpAttributes`), it will be failed to parse here. We need the overload.

Reply:
This change is created to be able to get not just VolumeSetAttributes in FreeIPA.